src/native/corehost/fxr/fx_resolver.h

```diff
@@ -9,41 +9,57 @@
 #include "fx_definition.h"
 
 class runtime_config_t;
-struct host_startup_info_t;
 
 class fx_resolver_t
 {
 public:
+    struct resolution_failure_info
+    {
+        fx_reference_t missing;
+        fx_reference_t incompatible_lower;
+        fx_reference_t incompatible_higher;
+        std::unique_ptr<fx_definition_t> invalid_config;
+    };
+
+public:
+    static StatusCode resolve_frameworks(
+        const pal::string_t& dotnet_root,
+        const runtime_config_t::settings_t& override_settings,
+        const runtime_config_t& app_config,
+        /*in_out*/ fx_definition_vector_t& fx_definitions,
+        resolution_failure_info& resolution_failure);
+
     static StatusCode resolve_frameworks_for_app(
-        const host_startup_info_t& host_info,
-        bool disable_multilevel_lookup,
+        const pal::string_t& dotnet_root,
         const runtime_config_t::settings_t& override_settings,
         const runtime_config_t& app_config,
-        fx_definition_vector_t& fx_definitions,
-        const pal::char_t* app_display_name = nullptr);
+        /*in_out*/ fx_definition_vector_t& fx_definitions,
+        const pal::char_t* app_display_name);
 
     static bool is_config_compatible_with_frameworks(
         const runtime_config_t& config,
         const std::unordered_map<pal::string_t, const fx_ver_t> &existing_framework_versions_by_name);
 
 private:
-    fx_resolver_t() = default;
+    fx_resolver_t(bool disable_multilevel_lookup, const runtime_config_t::settings_t& override_settings)
+        : m_disable_multilevel_lookup{disable_multilevel_lookup}
+        , m_override_settings{override_settings}
+    { }
```

Comment:
Something to consider. My guess here is that this is for some internal instance construction (that is, factory method). If that is true, it is likely to be done in a narrow set up scenarios. Ideally, construction would be in a single location and has few reuse needs, so one could create the default constructor and then set the needed fields. This approach avoids defining additional narrowly needed constructors.

Reply:
Okay.